wom/src/main/scala/wom/types/WomFileType.scala

```diff
@@ -1,28 +1,54 @@
 package wom.types
 
 import spray.json.JsString
-import wom.values.{WomFile, WomString}
+import wom.values.{WomFile, WomGlobFile, WomSingleFile, WomString}
 
 import scala.util.{Success, Try}
 
-case object WomFileType extends WomPrimitiveType {
+sealed abstract class WomFileType extends WomPrimitiveType
+
+case object WomSingleFileType extends WomFileType {
   val toDisplayString: String = "File"
 
   override protected def coercion = {
-    case s: String => WomFile(s)
-    case s: JsString => WomFile(s.value)
-    case s: WomString => WomFile(s.valueString)
+    case s: String => WomSingleFile(s)
+    case s: JsString => WomSingleFile(s.value)
+    case s: WomString => WomSingleFile(s.valueString)
+    case f: WomFile => f
+  }
+
+  override def add(rhs: WomType): Try[WomType] = rhs match {
+    case WomStringType => Success(WomSingleFileType)
+    case WomOptionalType(memberType) => add(memberType)
+    case _ => invalid(s"$this + $rhs")
+  }
+
+  override def equals(rhs: WomType): Try[WomType] = rhs match {
+    case WomSingleFileType => Success(WomBooleanType)
+    case WomStringType => Success(WomBooleanType)
+    case WomOptionalType(memberType) => equals(memberType)
```

Comment:
Is this true ? `WomSingleFileType == WomOptionalType(WomSingleFileType))` ?

Reply:
Interesting question. I don't know, and mainly refactored what was there. I haven't checked the spec, and the unit tests don't explicitly check for this case. This code is "covered by codecov", but only for the negative case via:

https://github.com/broadinstitute/cromwell/blob/bfa8378f0fa7b645df9c145e21f3b1a6f0c569c7/wdl/src/test/scala/wdl/expression/ValueEvaluatorSpec.scala#L31-L44

https://github.com/broadinstitute/cromwell/blob/bfa8378f0fa7b645df9c145e21f3b1a6f0c569c7/wdl/src/test/scala/wdl/expression/ValueEvaluatorSpec.scala#L472-L473

Reply:
okay, must be here for a reason. Maybe @cjllanwarne knows.
Anyway don't block the merging for this :)

Reply:
@kshakir @Horneth hmmmm, this looks a lot like a "this helps in some awkward situation and doesn't seem to break anything" kind of a line. 

I guess I'd like to see what breaks if we remove it and see if we can resolve that more appropriately (or maybe I've just forgotten what `equals` is used for in types). On the other hand if a tree falls down in the woods and nothing else seems to be broken...